chrome/content/zotero/integration/quickFormat.js

```diff
@@ -130,26 +175,46 @@ var Zotero_QuickFormat = new function () {
 					child.setAttribute("label", locatorLabel);
 					labelList.appendChild(child);
 				}
-
+				// If the focus leaves the citation properties panel, close it
+				panel.addEventListener("focusout", (_) => {
+					setTimeout(() => {
+						if (!panel.contains(document.activeElement)) {
+							panel.hidePopup();
+						}
+					});
+				});
 			}
 			
 			// Don't need to set noautohide dynamically on these platforms, so do it now
 			if(Zotero.isMac || Zotero.isWin) {
 				referencePanel.setAttribute("noautohide", true);
 			}
-		} else if (event.target === qfi.contentDocument) {
-			qfiWindow = qfi.contentWindow;
-			qfiDocument = qfi.contentDocument;
-			qfb.addEventListener("click", _onQuickSearchClick, false);
-			qfb.addEventListener("keypress", _onQuickSearchKeyPress, false);
-			qfe = qfiDocument.querySelector(".citation-dialog.editor");
-			qfe.addEventListener("drop", _onBubbleDrop, false);
-			qfe.addEventListener("paste", _onPaste, false);
-			if (Zotero_QuickFormat.citingNotes) {
-				_quickFormat();
-			}
+
+			// Resize whenever a bubble/input is added or removed
+			let resizeObserver = new MutationObserver(function (mutationsList) {
+				for (let mutation of mutationsList) {
+					if (mutation.type === 'childList') {
+						// Due to very odd mozilla behavior, calls to resize the window when
+						// dragging is happening make the next drag events after resize
+						// act strange (dragging doesn't happen, though dragstart fires but dragend doesn't)
+						// Resizing after delay in onBubbleDrop seems to not cause this issue
+						if (!bubbleDragged) {
+							_resizeWindow();
+						}
+					}
+				}
+			});
+			resizeObserver.observe(qfe, { childList: true });
```

Comment:
I found myself having to check if resizing needs to happen after pretty much anything changes within the editor: when an input is added, text is typed, an input is removed, a bubble is added, a page number is set/removed, etc. Because any of these actions can make the editor shorter or taller. So I thought it would be easier to handle it all in one place instead of calling `_resizeWindow` from multiple functions.

I did have a few places where `_resizeWindow` was called directly (somewhat defeating the purpose of this idea). So I made the observer listen to attributes as well, and now it's the only place that calls `_resize` whatever update happens within the editor.